conversation_reconstruction_dataflow/conversation_reconstruction/construct_utils/conversation_display.py

```diff
@@ -0,0 +1,107 @@
+            conv_mapping[action['parent_id']]
+        if action['type'] == 'COMMENT_RESTORATION':
+            conversations[conv_mapping[action['parent_id']]].append(action)
+            conv_mapping[action['id']] = conv_mapping[action['parent_id']]
+    return conversations, conv_mapping
+
+def update(snapshot, action):
+    Found = False
+    if action['type'] == 'COMMENT_REMOVAL':
+        for ind, act in enumerate(snapshot):
+            if action['parent_id'] in act['parent_ids']:
+                act['status'] = 'removed'
+                status = 'removed'
+                snapshot[ind] = act
+                Found = True
+    if action['type'] == 'COMMENT_RESTORATION':
+        for ind, act in enumerate(snapshot):
+            if action['parent_id'] in act['parent_ids']:
+                act['status'] = 'restored'
+                act['content'] = action['content']
+                status = 'restored'
+                snapshot[ind] = act
+                Found = True
+    if action['type'] == 'COMMENT_MODIFICATION':
+        found = False
+        for i, act in enumerate(snapshot):
+            if action['parent_id'] in act['parent_ids']:
+                found = True
+                pids = act['parent_ids']
+                act = copy.deepcopy(action)
+                act['parent_ids'] = pids
+                act['status'] = 'content changed'
+                status = 'content changed'
+                act['relative_replyTo'] = -1
+                act['parent_ids'][action['id']] = True
+                for ind, a in enumerate(snapshot):
+                    if a['id'] == action['replyTo_id']:
+                        act['relative_replyTo'] = ind
+                snapshot[i] = act
+                Found = True
+        if not(found):
+            act = copy.deepcopy(action)
+            act['status'] = 'just added'
+            status = 'just added'
+            act['relative_replyTo'] = -1
+            for ind, a in enumerate(snapshot):
+                if a['id'] == action['replyTo_id']:
+                    act['relative_replyTo'] = ind
+            act['parent_ids'] = {action['id'] : True}
+            snapshot.append(act)
+            Found = True
+    if action['type'] == 'COMMENT_ADDING' or action['type'] == 'SECTION_CREATION':
+        act = copy.deepcopy(action)
+        act['status'] = 'just added'
+        status = 'just added'
+        act['relative_replyTo'] = -1
+        Found = True
+        for ind, a in enumerate(snapshot):
+            if a['id'] == action['replyTo_id']:
+                act['relative_replyTo'] = ind
+        act['parent_ids'] = {action['id'] : True}
+        snapshot.append(act)
+    if not(Found): print(action)
+    return snapshot, status
+
+def display(snapshot):
+    print('======================SNAPSHOT=============================')
+    for ind, act in enumerate(snapshot):
+        print('ID:%d\nREVID:%s\nCONTENT:%s\nAUTHOR:%s\nTIMESTAMP:%s\nSTATUS:%s\nREPLY TO:%d\nREPLY TO ID:%s' \
```

Comment:
When you have a lot of arguments it is better to use explicit named params, see: https://stackoverflow.com/questions/15286401/print-multiple-arguments-in-python for examples.

Reply:
I think I'll put this into an issue, I want to integrate this part with the conversation viewer later so this part of code will be largely different.

Reply:
Since you'll be doing this part in the conversation viewer, I'll simply remove this?